java/ql/src/utils/model-generator/CaptureSourceModels.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Capture source models.
+ * @description Finds APIs that act as sources as they expose already known sources.
+ * @id java/utils/model-generator/sink-models
+ */
+
+import java
+private import Telemetry.ExternalAPI
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.dataflow.ExternalFlow
+private import ModelGeneratorUtils
+private import semmle.code.java.dataflow.internal.FlowSummaryImplSpecific
+private import semmle.code.java.dataflow.internal.FlowSummaryImpl
+private import semmle.code.java.dataflow.internal.DataFlowImplCommon
+private import semmle.code.java.dataflow.internal.DataFlowNodes::Private
+
+class FromSourceConfiguration extends TaintTracking::Configuration {
+  FromSourceConfiguration() { this = "FromSourceConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { sourceNode(source, _) }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Callable c |
+      sink instanceof ReturnNodeExt and
+      sink.getEnclosingCallable() = c and
+      c.isPublic() and
+      c.fromSource()
+    )
+    or
+    exists(MethodAccess c | sink.asExpr() = c.getAnArgument())
+  }
+
+  override DataFlow::FlowFeature getAFeature() {
+    result instanceof DataFlow::FeatureHasSinkCallContext
+  }
+}
+
+string asOutput(DataFlow::Node node) {
+  if node instanceof ReturnNodeExt
+  then result = "ReturnValue"
```

Comment:
This is not entirely correct. What's the ambition level here? For the regular `ReturnNode` subclass of `ReturnNodeExt` then this is fine, but `ReturnNodeExt` also covers post-update nodes of parameters, i.e. places where taint is written to a parameter.
```suggestion
  if node instanceof ReturnNodeExt
  then
    result = "ReturnValue" and node.(ReturnNodeExt).getKind() instanceof ValueReturnKind or
    result = "Parameter[" + node.(ReturnNodeExt).getKind().(ParamUpdateReturnKind).getPosition() + "]"
```

Reply:
See also `string asOutput(Callable api, ReturnNodeExt node)` in `CaptureSummaryModels.ql` where `ReturnNodeExt` is handled correctly.

Reply:
As suggested in the other comments, we should take a more holistic approach to supporting to handle the `Parameter` case. For the `Argument` case, I did apply your suggestion and added a few tests to capture this.